### PR TITLE
[FTAMS1020202] Nina van den Braak

### PR DESCRIPTION
**Pull request by ninsajala**
Iteration 1,2,3 done, now over to War

### CODE REVIEW

File: src/viking.js [hunk 1]
```diff
@@ -1,11 +1,125 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health,
+            this.strength = strength
```
Comment (wax-bear) [inline src/viking.js:5]: The indentation seems off here.

File: src/viking.js [hunk 2]
```diff
@@ -1,11 +1,125 @@
...
+    }
+
+    attack() {
+        return this.strength
+
```
Comment (wax-bear) [inline src/viking.js:10]: Please be mindful about the use of empty lines. Does this empty line improve the readability of the code?

File: src/viking.js [hunk 3]
```diff
@@ -1,11 +1,125 @@
...
+    addViking(Viking) {
+        this.vikingArmy.push(Viking)
+    }
+
+    addSaxon(Saxon) {
+        this.saxonArmy.push(Saxon)
+    }
+
+    vikingAttack() {
+        let saxon = randomIndex(this.saxonArmy)
```
Comment (wax-bear) [inline src/viking.js:73]: Will the value of the variable saxon be reasigned? If not, would, perhaps, a const be more apt here?

File: src/viking.js [hunk 4]
```diff
@@ -1,11 +1,125 @@
...
+        let viking = randomIndex(this.vikingArmy)
```
Comment (wax-bear) [inline src/viking.js:74]: Nice you are making use of a function here. Could the name, perhaps, be improved? What is this function ultimately doing? Are we getting something particular? If so, could that be processed in the function name?

File: src/viking.js [hunk 5]
```diff
@@ -1,11 +1,125 @@
...
+            return 'Saxons have fought for their lives and survived another day...'
+        }
+        if (this.vikingArmy.length > 0 && this.saxonArmy.length > 0) {
+            return 'Vikings and Saxons are still in the thick of battle.'
+
+        }
+    }
+}
+
+function randomIndex(array) {
```
Comment (wax-bear) [inline src/viking.js:109]: As discussed above, the naming could be altered. For example "getRandomWarrior" could do. Wouldn't it be better to make that function a method of the War class?

File: src/viking.js [hunk 6]
```diff
@@ -1,11 +1,125 @@
...
+    return array[Math.floor(Math.random() * array.length)]
+}
+
+function cleanDeads(array) {
```
Comment (wax-bear) [inline src/viking.js:113]: Same question regarding function/class method applies to cleanDeads.

File: src/viking.js [hunk 7]
```diff
@@ -1,11 +1,125 @@
...
+    return array.filter(item => item.health > 0)
```
Comment (wax-bear) [inline src/viking.js]: Imagine the army which is generated is big, contains many items. Filter, in that case, would be more costly than simply splicing the particular warrior based on the index which is used before to select one from the viking/saxon warriors array.

https://stackoverflow.com/a/44435418/3563737

A workaround would be to generate the index outside of the previous function and to store it in a variable. Then, the index can be submitted as a second argument to that function, and then later also be used here to splice the particular element.

File: src/viking.js (hunk 1 above)
Comment (ninsajala) [reply src/viking.js:5]: Check

File: src/viking.js (hunk 2 above)
Comment (ninsajala) [reply src/viking.js:10]: Will keep this in mind more, I tend to forget when I'm trying to solve the iterations. (Any extensions that can help me with it?)

File: src/viking.js (hunk 3 above)
Comment (ninsajala) [reply src/viking.js:73]: Yes, it's reassigned when the saxon is receiving damage

File: src/viking.js (hunk 4 above)
Comment (ninsajala) [reply src/viking.js:74]: Check

**Review by wax-bear — COMMENTED** [review]
Good stuff Nina!
Please check out my comments for potential improvements.
Please reach out if you have any questions.

File: src/viking.js (hunk 2 above)
Comment (wax-bear) [reply src/viking.js:10]: I am not aware of such an extension unfortunately.

**Pull request closed without merging**